### DISCUSSION TITLE
Filament filter on value in joined table

### DISCUSSION THREAD

**Discussion by erwindeclerck** [filamentphp/filament#9571] [General]
Hi,

I'm trying to create a filter within the table builder on a value from a joined table.
Main base table is 'hosts' and the joined table is 'saldi', which is a oneonone table linked trough a foreign key saldi.host to hosts.id.
Within the saldi table there is a column (assigned) with an integer varying from zero to about 10.
I want to filter rows on which this value is > 2.

                Filter::make('test')->query(function($query){ $query->leftJoin('saldi',function(JoinClause $join){
                    $join->on('hosts.id','=','saldi.host')->where('saldi.assigned','>',2);
                }) ;})->toggle(),

The build query is:

"select * from `hosts` left join `saldi` on `hosts`.`id` = `saldi`.`host` **and** `saldi`.`assigned` > ?" 
array:1 [
  0 => 2
]

Unfortunately the querybuilder provides 'and' in stead of 'where', in which case the query would work correct.

Can anybody tell me what I'am doing wrong?

kind regards,
Ewin

**Comment by sinnbeck** [comment]
Try
```php
            Filter::make('test')->query(function($query){ $query->leftJoin('saldi',function(JoinClause $join){
                $join->on('hosts.id','=','saldi.host');
            })->where('saldi.assigned','>',2) ;})->toggle(),
```

**Reply by erwindeclerck** [reply]
Thx!
But I tried this one also and then I get the following result:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'saldi.assigned' in 'where clause'
SELECT count(*) AS aggregate FROM `hosts` WHERE `id` > 10 AND (`province` = NULL AND `published` = 1 AND `saldi`.`assigned` > 2)

It seems that the where() needs an other variable to be given in order to be able to modify that join... 
I had the impression that it should be possible to inject an other global variable $join into the closure, but that doesn't seem to exist....

**Reply by erwindeclerck** [reply]
Sorry I pasted the query that was combined with other filters.
This filter only gives this query:

SELECT
  count(*) AS aggregate
FROM
  `hosts`
WHERE
  `id` > 10
  AND (`saldi`.`assigned` > 2)

The problem now is that there is no join happening at all...
In the original version, chaining the where-claus after the on-clause, the where-clause is mixed up with the on-clause and in stead of WHERE, AND is given...

**Reply by mbise1993** [reply]
Did you ever figure this out? I'm also seeing the issue you mentioned where my `JOIN` is not applied to the query that gets executed

**Reply by erwindeclerck** [reply]
See https://laracasts.com/discuss/channels/laravel/filter-table-on-joined-table-column?page=1&replyId=913504
for 'some kind of a solution'...

**Comment by LucExitable** [comment]
Is this addressed in newer versions? Joins in a filter sounds like a common thing to do, but due to this issue is practically not possible...

**Reply by memdufaizan** [reply]
```
return $table
            //->query(hosts::query())
            ->query(hosts::query()->join('saldi', 'hosts.id', '=', 'saldi.id'))
            ->columns([
                //..
            ])
            ->filters([
                  Filter::make('test')->query(function($query){ $query->leftJoin('saldi',function(JoinClause $join){
                      $join->on('hosts.id','=','saldi.host');
                  })->where('saldi.assigned','>',2) ;})->toggle(),
             ])

```
when you used join query in filter then also add join query in table also.
